src/MongoDB/Session.c

```diff
@@ -764,6 +770,39 @@ static HashTable* php_phongo_session_get_debug_info(phongo_compat_object_handler
 		ADD_ASSOC_NULL_EX(&retval, "server");
 	}
 
+	if (intern->client_session) {
+		ADD_ASSOC_BOOL_EX(&retval, "inTransaction", mongoc_client_session_in_transaction(intern->client_session));
+	} else {
+		ADD_ASSOC_NULL_EX(&retval, "inTransaction");
+	}
+
+	if (intern->client_session) {
+		const char* state = php_phongo_get_transaction_state_string(mongoc_client_session_get_transaction_state(intern->client_session));
+		if (state) {
+			ADD_ASSOC_STRING(&retval, "transactionState", state);
+		} else {
+			ADD_ASSOC_NULL_EX(&retval, "transactionState");
+		}
+	} else {
+		ADD_ASSOC_NULL_EX(&retval, "transactionState");
+	}
+
+	if (intern->client_session) {
+		mongoc_transaction_opt_t* txn_opts = mongoc_session_opts_get_transaction_opts(intern->client_session);
```

Comment:
I refactored the helper function `php_phongo_transaction_options_to_zval` and freed `opts` in it as you said to do, but I wasn't able to verify the memory leak that you noted in https://jira.mongodb.org/browse/PHPC-1879 since the `session-getTransactionOptions-001.phpt` test gets skipped when I run it. How should I test that?

Reply:
Why is `session-getTransactionOptions-001.phpt` skipped? libmongoc should definitely have crypto available and your local environment should be a replica set (if only a single primary) running at least MongoDB 4.0. Both of those should satisfy the SKIPIF.

If you happened to wipe your data directory for `mongod`, did you reinitialize the replica set after re-launching it? The command we used when first setting it up was [`rs.initiate()`](https://docs.mongodb.com/manual/reference/method/rs.initiate/). And recall that `mongod` should be started with the [`--replSet`](https://docs.mongodb.com/manual/reference/program/mongod/#std-option-mongod.--replSet) command line option.

If `mongod` is definitely a replica set, does your `MONGODB_URI` environment variable still specify the replica set name in the connection string?

Reply:
Hmm yes, I think I am following everything correctly with `rs.initiate()`, `--replSet`, and the `MONGODB_URI`. The test is being skipped because of the  `<?php skip_if_no_transactions(); ?>`  (the message says "topology does not support transactions").